src/components/shared/PinchZoomView.tsx

```diff
@@ -0,0 +1,100 @@
+import { ImageProps, SafeAreaView, TouchableOpacity, View } from 'react-native';
+import ImageViewer, { ImageViewerPropsDefine } from 'react-native-image-zoom-viewer';
+import React, { ReactElement } from 'react';
+import { Ionicons } from '@expo/vector-icons';
+import styled from 'styled-components/native';
+
+export interface ImageI extends HeaderProps {
```

Comment:
`ImageI` looks awkward. Better naming would be like `ImageUrl` or something meaningful. Also, I don't get why you are extending `HeaderProps` and making `Image` type bit complicating.

Since `HeaderProps` is used completely in different elements, try to separate them.

Reply:
Try to use `IImageInfo` defined in `react-native-image-zoom-viewer` for typings. Please add it to props for `PinchZoomView`.

```
images: IImageInfo[];
title: string;
subtitle: string;
onClosePressed: () => void;
...
```

The suggestion for typing is as above.

Reply:
`ImageInfo` can accept additional information of images as title, subtitle for header.
I change it to extending `IImageInfo`

```
export interface ImageInfo extends IImageInfo {
  title: string;
  subtitle: string;
}
```